### PR TITLE
Update file uploads for ESM

### PR DESCRIPTION
**Pull request by FrozenKiwi**
The multer used in fileUploads generated a 'require' statement, which is not legal in ESM packages.

In a local test, simply changing from require to default import fixed the issue

### All Submissions:

- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [X] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [X] This PR is associated with an existing issue?

**Closing issues**

`closes 1278`

### CODE REVIEW

**Comment by FrozenKiwi** [comment]
Ok, this PR became much bigger than initial anticipated.

When loading into an ESM-native node project, the `import * as mutler from 'multer'` syntax doesn't work, because according to the spec this is supposed to be a namespace which doesn't support direct function calls.  See https://www.typescriptlang.org/tsconfig#esModuleInterop

The fix is to add `esModuleInterop`, which unfortunately then breaks all the tests as they rely on the older method of interop.

After spending an afternoon trying to get this working without forcing the `esModuleInterop` flag upstream, I can't see another way to get this working.

I'd understand if you drop this PR, however as ESM is the future it might pay to move this 1 step closer to compliance

File: packages/cli/src/routeGeneration/templates/express.hbs [hunk 1]
```diff
@@ -19,10 +19,9 @@ const promiseAny = require('promise.any');
 import { iocContainer } from '{{iocModule}}';
 import { IocContainer, IocContainerFactory } from '@tsoa/runtime';
 {{/if}}
-import type { RequestHandler } from 'express';
```
Comment (FrozenKiwi) [inline packages/cli/src/routeGeneration/templates/express.hbs:22]: minor unrelated suggestion, only importing the type makes the intent of the file clearer

File: packages/cli/src/routeGeneration/templates/koa.hbs [hunk 2]
```diff
@@ -24,7 +24,7 @@ import { iocContainer } from '{{iocModule}}';
 import { IocContainer, IocContainerFactory } from '@tsoa/runtime';
 {{/if}}
 import type { Middleware } from 'koa';
-import * as KoaRouter from '@koa/router';
```
Comment (FrozenKiwi) [inline packages/cli/src/routeGeneration/templates/koa.hbs:27]: I don't know Koa so can't test this, but I think it's covered in the tests and I assume it's better to have a consistent import usage across the code base?

File: tests/tsconfig.json [hunk 3]
```diff
@@ -9,6 +9,7 @@
     "downlevelIteration": true,
     "experimentalDecorators": true,
     "emitDecoratorMetadata": true,
+    "esModuleInterop": true,
```
Comment (FrozenKiwi) [inline tests/tsconfig.json:12]: note, change required to compile tests with `import XXX from 'XXX'`

**Comment by ClementValot** [comment]
Isn't this 

`{{#if esm}}
import multer from 'multer';
{{else}}
const multer = require('multer');
{{/if}}`

an easier fix?

**Comment by FrozenKiwi** [comment]
Possibly?

The rest of the imports are `import`, not `require`, and there didn't seem to be a strong advantage to leaving this one tag in an obsolete format.

https://github.com/lukeautry/tsoa/blob/057c57b951d01f32c49a5ae801a29a46162b6696/packages/cli/src/routeGeneration/templates/express.hbs#L22

**Comment by FrozenKiwi** [comment]
I guess it wouldn't have then been necessary to modify the `tsconfig`.  Honestly, it didn't occur to me, which is a little silly.  Up to you which way you want to fix this.

**Comment by FrozenKiwi** [comment]
See new PR

**Pull request closed without merging**